src/Compilers/CSharp/Portable/BoundTree/DecisionTree.cs

```diff
@@ -53,7 +53,7 @@ public static DecisionTree Create(BoundExpression expression, TypeSymbol type, S
                 expression = new BoundLocal(expression.Syntax, temp, null, type);
             }
 
-            if (expression.Type.CanBeAssignedNull())
```

Comment:
The code there is correct as written - you can't have `case null` if you're switching on a value whose type is an unconstrained type parameter, precisely because `null` can't be assigned to that type. However, the code would probably be clearer with the test there changed as well (and more efficient at runtime in the case that the input is actually null). So I'll change it. I'll also introduce a helper for `CanContainNull` for clarity.